packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -98,9 +98,32 @@ export async function load_node({
 				let response;
 
 				if (/^[a-zA-Z]+:/.test(url)) {
-					// external fetch
-					const request = new Request(url, /** @type {RequestInit} */ (opts));
-					response = await options.hooks.serverFetch.call(null, request);
+					// possibly external fetch
+					if (typeof request.host !== 'undefined') {
+						const { hostname: fetchHostname } = new URL(url);
+						const [serverHostname] = request.host.split(':');
+
+						// allow cookie passthrough for "same-origin"
+						// if SvelteKit is serving my.domain.com:
+						// -        domain.com WILL NOT receive cookies
+						// -     my.domain.com WILL receive cookies
+						// -    amy.domain.com WILL NOT receive cookies
```

Comment:
this line and the next line seem a bit duplicative. we probably don't need both. I wonder if this line is a typo since it's so similar to "my"?

Reply:
Not a typo - it's showing a case where `fetchHostname.endsWith(serverHostname)` without prepending a dot would incorrectly pass cookies.

Reply:
I think I'd remove this line since it's already covered below where you say:
> leading dot prevents mydomain.com matching domain.com